web/models/course/course_helper.ex

```diff
@@ -0,0 +1,31 @@
+defmodule CoursePlanner.CourseHelper do
+  @moduledoc """
+  This module provides custom functionality for controller over the model
+  """
+  use CoursePlanner.Web, :model
+  import Ecto.DateTime, only: [utc: 0]
+
+  alias CoursePlanner.Repo
+  alias CoursePlanner.Course
+
+  def delete_handler(course) do
+    case course.status do
+      "Planned" -> hard_delete_course(course)
+      _         -> soft_delete_course(course)
+    end
+  end
+
+  defp soft_delete_course(course) do
+    changeset = change(course, %{status: "Deleted", deleted_at: utc()})
```

Comment:
As I wrote above, now i have big doubt wether a deleted item should be at all kept. since we have no plan of giving access to the deleted data. plus is not so informative. we can even make it easier by generally dropping soft delete and just hard delete any item